x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/aggregations/metrics/AggregateMetricsAggregatorsRegistrar.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.aggregatemetric.aggregations.metrics;
+
+import org.elasticsearch.search.aggregations.metrics.AvgAggregationBuilder;
+import org.elasticsearch.search.aggregations.metrics.MaxAggregationBuilder;
+import org.elasticsearch.search.aggregations.metrics.MetricAggregatorSupplier;
+import org.elasticsearch.search.aggregations.metrics.MinAggregationBuilder;
+import org.elasticsearch.search.aggregations.metrics.MinMaxAggregatorSupplier;
+import org.elasticsearch.search.aggregations.metrics.SumAggregationBuilder;
+import org.elasticsearch.search.aggregations.metrics.ValueCountAggregationBuilder;
+import org.elasticsearch.search.aggregations.metrics.ValueCountAggregatorSupplier;
+import org.elasticsearch.search.aggregations.support.ValuesSourceRegistry;
+import org.elasticsearch.xpack.aggregatemetric.aggregations.support.AggregateMetricsValuesSource;
+import org.elasticsearch.xpack.aggregatemetric.aggregations.support.AggregateMetricsValuesSourceType;
+
+/**
+ * Utility class providing static methods to register aggregators for the aggregate_metric values source
+ */
+public class AggregateMetricsAggregatorsRegistrar {
+
+    public static void registerSumAggregator(ValuesSourceRegistry.Builder builder) {
+        builder.register(
+            SumAggregationBuilder.NAME,
+            AggregateMetricsValuesSourceType.AGGREGATE_METRIC,
+            (MetricAggregatorSupplier) (name, valuesSource, formatter, context, parent, metadata) -> new AggregateMetricBackedSumAggregator(
+                name,
+                (AggregateMetricsValuesSource.AggregateDoubleMetric) valuesSource,
+                formatter,
```

Comment:
Just a heads up - I changed `MetricAggregatorSupplier` to take a `ValuesSourceConfig` here instead of separate `ValuesSource` and `DocValuesFormat` args.  I've cleaned up VSConfig so it can be used as an actual config object rather than having to pick it apart in the factories.

Reply:
Thanks @not-napoleon. I will fix that